2018-12-05/path_fraction.py

```diff
@@ -0,0 +1,121 @@
+"""
+12/05/2018: Problem 6:
+Given a sequence of points where consecutive points
+are connected by line segments to make a path,
+return the point along the path that corresponds
+to a given percentage along the path.
+
+Author: Eugene Ma
+"""
+
+import collections
+
+Point = collections.namedtuple('Point', ['x', 'y'])
+
+def get_dist(pt1, pt2):
+    import math
```

Comment:
Cool solution! If you don't mind, please [move imports to the top of the file](https://stackoverflow.com/a/1188672/6243352) and I'll merge the commits. Thanks!

Reply:
OK, running the code, I see you've drawn the line but is a solution produced? My understanding is that you're given, say, `[(5, 2), (6, 4), (7, 6)]` and `0.5` (meaning 50%) and should return `(6, 4)` because it's exactly 50% of the way through the line. This example is trivial, though; for `[(5, 5), (6, 2), (7, 2)]` and `0.5`, the answer should be something like `(5.8, 2.1)` because the `(5, 5) -> (6, 2)` line is a bit longer than the `(6, 2), (7, 2)` line. What do you think?